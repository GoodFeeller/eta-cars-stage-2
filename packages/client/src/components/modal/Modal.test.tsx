import { render, screen } from '@testing-library/react'
import Modal from './Modal'
import userEvent from '@testing-library/user-event'
const setCoin = jest.fn((_a: any) => null)
const setShow = jest.fn((_a: any) => null)

jest.mock('react', () => {
  const actual = jest.requireActual('react')
  return {
    __esModule: true,
    ...actual,
    useContext: () => {return {
      setCoin,
      setShow
    }
  }
}})
describe('Modal testing', () => {
  test('Snapshot test', () => {
    render(
      <Modal>
        <div />
      </Modal>,
    )
    const modal = screen.queryByTestId('modal')
    expect(modal).toMatchSnapshot()
  })
  test('Context test', () => {
    render(
      <Modal>
        <div />
      </Modal>,
    )
    const modal = screen.queryByTestId('modal') as HTMLElement
    userEvent.click(modal)
    expect(setCoin).toHaveBeenCalled()
    expect(setShow).toHaveBeenCalled()
  })
})
