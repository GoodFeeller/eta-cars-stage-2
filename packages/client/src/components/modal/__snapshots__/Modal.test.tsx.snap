// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal testing Snapshot test 1`] = `Promise {}`;
